pyfr/integrators/dual/phys/controllers.py

```diff
@@ -47,8 +51,21 @@ def advance_to(self, t):
             raise ValueError('Advance time is in the past')
 
         while self.tcurr < t:
+
+            if self.tcurr + 2*self._dt < t < self.tcurr + 3*self._dt:
+                dt = 0.5*(t - self.tcurr)
+            elif self.tcurr + self._dt < t < self.tcurr + 2*self._dt:
+                dt = t - self.tcurr
+            else:
+                dt = self._dt
+
+            if self.pseudointegrator.dt != dt:
```

Comment:
I 'd just overwrite without the if check

Reply:
Overwriting dt every step  also means modifying dt/dtau ratio every step. The `dtau` fields and/or (`pseudo-dt-min-mult` and `pseudo-dt-max-mult`) will be changed every time-step. I would guess that will be expensive.